src/core/tsi/ssl_transport_security.cc

```diff
@@ -1732,7 +1757,11 @@ tsi_result tsi_create_ssl_client_handshaker_factory_with_options(
     tsi_ssl_handshaker_factory_unref(&impl->base);
     return result;
   }
-  SSL_CTX_set_verify(ssl_context, SSL_VERIFY_PEER, nullptr);
+  if (options->server_verification_option == GRPC_SSL_SKIP_SERVER_VERIFICATION) {
```

Comment:
Unfortunately, the design of transport security is that it is independent module and namespace from gRPC. Thus ssl_transport_security cannot depend on grpc_security.h here and we should not use GRPC_SSL_SKIP_SERVER_VERIFICATION directly.

You can refer how GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE converts to TSI_DONT_REQUEST_CLIENT_CERTIFICATE, as an example. This is a little bit inconvenient though.

Reply:
Changed.